model/storage/src/main/java/org/keycloak/storage/adapter/AbstractUserAdapterFederatedStorage.java

```diff
@@ -429,4 +430,19 @@ public Streams(final KeycloakSession session, final RealmModel realm, final Comp
             super(session, realm, storageProviderModel);
         }
     }
+
+    private MultivaluedHashMap<String, String> getFederatedUserAttributes() {
+        if (federatedAttributes == null) {
+            federatedAttributes = getFederatedStorage().getAttributes(realm, this.getId());
+        }
+        return Optional.ofNullable(federatedAttributes).map(MultivaluedHashMap::new).orElse(new MultivaluedHashMap<>());
```

Comment:
Nitpick: I see that if `federatedAttributes` is null, an empty `MultivaluedHashMap` is returned. Instead, I would check only once if `getFederatedStorage().getAttributes(realm, this.getId())` returns `null`, and then initialize an empty `MultivaluedHashMap` as this would prevent multiple calls if this would return `null` (which isn't the case for the JPA implementation, but who knows about others).

Reply:
Yeah, I'm assuming the storage does not return null.